### PR TITLE
(PUP-8073) Remove bolt functions as part of BOLT-185

### PR DESCRIPTION
**Pull request by MikaelSmith**
They required a Bolt executor, so have been moved to Bolt.

### CODE REVIEW

File: spec/unit/puppet_pal_2pec.rb [hunk 1]
```diff
@@ -755,25 +696,6 @@ def arubyfunc
...
-      it 'a plan in a module can be called with run_plan' do
-        testing_env_dir # creates the structure
-        result = Puppet::Pal.in_environment('pal_env', envpath: environments_dir, facts: node_facts) do |ctx|
-          ctx.with_script_compiler {|c| c.evaluate_string('run_plan("a::aplan")') }
-        end
-        expect(result).to eq("a::aplan value")
-      end
-
-      # Note that when run_plan moves to bolt, this needs to move there
-      it 'the envpath can have multiple entries - that are searched for the given env' do
```
Comment (MikaelSmith) [inline spec/unit/puppet_pal_2pec.rb]: I'm going to adapt a few of these tests to run without `run_plan`, as they exercise other functionality in the PAL API.

File: spec/unit/puppet_pal_2pec.rb [hunk 2]
```diff
@@ -112,18 +104,6 @@
             expect { ctx.with_script_compiler {|c| c.evaluate_string('$a') }}.to raise_error(/Unknown variable: 'a'/)
           end
         end
-
-        it 'instantiates definitions in the given code string' do
```
Comment (hlindberg) [inline spec/unit/puppet_pal_2pec.rb]: This test shoud remain, but be changed to not run a plan, but some other definition - like a function.
Simply change to "function run_me()" and use "call_function". It is not testing plan per se.

File: spec/unit/puppet_pal_2pec.rb [hunk 3]
```diff
@@ -134,20 +114,6 @@
           end
           expect(result).to eq(10)
         end
-
-        it 'instantiates definitions in the given code string' do
```
Comment (hlindberg) [inline spec/unit/puppet_pal_2pec.rb]: Same thing here - should change to being a test to call a function - change 'plan' to 'function'

File: spec/unit/puppet_pal_2pec.rb [hunk 4]
```diff
@@ -71,14 +71,6 @@
         end
         expect(result).to eq(10)
       end
-
-      it '"run_plan" runs a plan with specified content in a manifest' do
```
Comment (hlindberg) [inline spec/unit/puppet_pal_2pec.rb]: ok to remove together with the impl, but bolt must be in sync and use `call_function('run_plan',...)` in order to continue working.

File: lib/puppet_pal.rb [hunk 5]
```diff
@@ -310,24 +310,6 @@ def self.evaluate_script_manifest(manifest_file)
...
 
-  # Runs the given named plan passing arguments by name in a hash.
-  # @param plan_name [String] the name of the plan to run
-  # @param plan_args [Hash] arguments to the plan - a map of plan parameter name to value, defaults to empty hash
-  # @param manifest_file [String] a Puppet Language file to load and evaluate before running the plan, mutually exclusive with code_string
-  # @param code_string [String] a Puppet Language source string to load and evaluate before running the plan, mutually exclusive with manifest_file
-  # @return [Object] returns what the evaluated plan returns
-  # @deprecated Use {#with_script_compiler} and then `call_function('run_plan', plan_name, plan_args)` on the given compiler - to be removed in 1.0 version
-  #
-  def self.run_plan(plan_name,
```
Comment (hlindberg) [inline lib/puppet_pal.rb]: ok to remove but bolt must be in sync and use call_function instead

Comment (MikaelSmith) [reply lib/puppet_pal.rb]: Yes, that's the plan.

File: spec/unit/puppet_pal_2pec.rb (hunk 2 above)
Comment (MikaelSmith) [reply spec/unit/puppet_pal_2pec.rb]: Done.

**Comment by MikaelSmith** [comment]
https://github.com/puppetlabs/bolt/pull/166 is the necessary bolt work. Looks like it can be merged without the changes in this PR.

**Pull request merged**